sotodlib/core/metadata/loader.py

```diff
@@ -71,229 +85,408 @@ def load_raw(self, spec_list, request, detdb=None):
+
+        # Lookup.
+        try:
+            index_lines = man.match(request, multi=True, prefix=dbpath)
+        except Exception as e:
+            text = str(e)
+            raise RuntimeError(
+                'An exception was raised while decoding the following spec:\n'
+                + '  ' + str(spec) + '\n'
+                + 'with the following request:\n'
+                + '  ' + str(request) + '\n'
+                + 'The exception is:\n  %s' % text)
+
+        # Load each index_line.
+        results = []
+        for index_line in index_lines:
+            logger.debug(f'Processing index_line={index_line}')
+            # Augment the index_line with info from the request; if
+            # the request and index_line share a key but conflict on
+            # the value, we don't want this item.
+            skip_this = False
+            for k in request:
+                if k in index_line:
+                    if request[k] != index_line[k]:
+                        skip_this = True
+            if skip_this:
+                continue
+            index_line.update(request)
+            loader = spec.get('loader', None)
+            if loader is None:
+                loader = index_line.get('loader', REGISTRY['_default'])
+            try:
+                loader_class = REGISTRY[loader]
+            except KeyError:
+                raise RuntimeError('No metadata loader registered under name "%s"' % loader)
+
+            loader_object = loader_class()  # pass obs info?
+            mi1 = loader_object.from_loadspec(index_line)
+
+            # Restrict returned values according to the specs in index_line.
+
+            if isinstance(mi1, ResultSet):
+                # For simple tables, the restrictions can be
+                # integrated into the table, to be dealt with later.
+                det_restricts = _filter_items('dets:', index_line, remove=False)
+                mask = np.ones(len(mi1), bool)
+                keep_cols = list(mi1.keys)
+                new_cols = []
+                for k, v in det_restricts.items():
+                    if k in mi1.keys:
+                        mask *= (mi1[k] == v)
+                    else:
+                        new_cols.append((k, v))
+                a = mi1.subset(keys=keep_cols, rows=mask)
+                mi2 = ResultSet([k for k, v in new_cols],
+                                [[v for k, v in new_cols]] * len(a))
+                mi2.merge(a)
+
+            elif isinstance(mi1, core.AxisManager):
+                # For AxisManager results, the dets axis *must*
+                # reconcile 1-to-1 with some field in det_info, and
+                # that may be used to toss things out based on
+                # index_line.
+                det_restricts = _filter_items('dets:', index_line, remove=True)
+                dets_key = 'name'
+                new_dets, i_new, i_info = core.util.get_coindices(mi1.dets.vals, det_info['name'])
```

Comment:
I think there's a bug around here but don't know the right fix. Trying to run this with some previously working cal_dbs, I get an error about looking up "name." Printing the det_info ResultSet gets me
```
ResultSet<[readout_id,smurf_band,smurf_channel,res_frequency,detset], 878 rows>
```
If I change `det_info['name']` to `det_info['readout_id']` everything works and I get the correct calibration data loaded.

Reply:
Ah ... yes, this was a stray.  It should be readout_id now, as you suggest.  Fixed.